src/audio/types.h

```diff
@@ -244,6 +244,53 @@ class Bitrate {
 
 QDebug operator<<(QDebug dbg, Bitrate arg);
 
+// The BarLength is measured in beats and provides information
+// about the number expected within a single bar or phrase.
+class BarLength {
+  public:
+    using value_t = std::uint32_t;
+
+    static BarLength valueFromUInt(unsigned int value) {
+        VERIFY_OR_DEBUG_ASSERT(value >= std::numeric_limits<value_t>::min() &&
+                value <= std::numeric_limits<value_t>::max()) {
+            return BarLength(0);
```

Comment:
its not, is `isValid()`.

Reply:
What is the use case?

Reply:
I am not sure to be honest - the first version was falling back to the default value. However, I guess it semantically makes sense for a `BeatsPerBar` to be invalid if instantiated from an in out of bound. This not actively used tho.

Reply:
Than I would prefer to either use a the valid range or 0 as invalid and rejects everything else. 

The question is open how to deal with the "defaultet 4/4" state not sure if this a database only value or if we need to be able to transport this information with the type.

Reply:
> Than I would prefer to either use a the valid range or 0 as invalid and rejects everything else.

I can't follow you here. Can you rephrase?

> The question is open how to deal with the "defaultet 4/4" state not sure if this a database only value or if we need to be able to transport this information with the type.

I think its a legacy assumption that should not be in the type tbh. The less assumption a type makes, the more versatile it becomes. Only make the 4/4 assumption when upgrading from the legacy format to the new one.

Reply:
Same for validity tbh. If we don't use it currently, don't introduce the notion (but once you do need it, don't work around the lack of it by breaking the encapsulation (manually checking checking the range)).

Reply:
> I can't follow you here. Can you rephrase?

The idea was to default to zero in the invalid case, but I just noticed that the other types also don't do it. So my comment is void.

Reply:
> Only make the 4/4 assumption when upgrading from the legacy format to the new one.

Thats the point. We must not loose the "not decided state". I don't mind if the type can hold this state or not as long the business logic does it somehow.

Reply:
I don't fully agree with that. When/How would that decision be made? Manual intervention from the user for every single track is not an option because its infeasible with any reasonably sized library.

Reply:
The decision is made, once the user has adjusted the bars. There is no requirement to do this but when playing track the user should be able to know whether it is a confirmed 4/4 track where he can rely on or a guessed one. 
If users are fine to use the guessed one, it's fine. But at least they can see the difference and may filter and double check the guessed versions. 
Once we have a bar detection in place, we will probably have three states "guessed," "Detected by detector version x" and "manually confirmed". The detector can than probably run only on the guessed bar tracks. 
I use currently the best lock for this. But there is a use case to guess and detect bars on locked beat grids.

Reply:
> Once we have a bar detection in place, we will probably have three states "guessed," "Detected by detector version x" and "manually confirmed".

It sounds like something that will require to commit on a new proto, which we have manage to skip for that piece of work, so I would say this is probably not something to worry about for now. 

It turns out that this method isn't currently used, so I removed it. This way we don't make that a blocker.